pyannote/audio/tasks/segmentation/mixins.py

```diff
@@ -129,7 +130,16 @@ def default_metric(
         """Returns macro-average of the area under the ROC curve"""
 
         num_classes = len(self.specifications.classes)
-        return AUROC(num_classes, pos_label=1, average="macro", compute_on_step=False)
+        if self.specifications.problem == Problem.BINARY_CLASSIFICATION:
+            return BinaryAUROC(average="macro", compute_on_step=False)
```

Comment:
While we are at it, I would also use [`compute_on_cpu`](https://torchmetrics.readthedocs.io/en/v0.11.0/pages/overview.html#advanced-metric-settings) (added in 0.8.0) for `*AUROC` metrics because this particular metrics have to store all validation `y_pred` and `y` in memory before computing the actual metric and it might cause memory issue for large validation sets.